contracts/vaults/BaseVaults/RibbonThetaVault.sol

```diff
@@ -408,12 +411,23 @@ contract RibbonThetaVault is RibbonVault, RibbonThetaVaultStorage {
 
         emit OpenShort(newOption, lockedBalance, msg.sender);
 
-        VaultLifecycle.createShort(
-            GAMMA_CONTROLLER,
-            MARGIN_POOL,
-            newOption,
-            lockedBalance
-        );
+        uint256 optionsMintAmount =
+            VaultLifecycle.createShort(
+                GAMMA_CONTROLLER,
+                MARGIN_POOL,
+                newOption,
+                lockedBalance
+            );
+
+        address _optionsPurchaseQueue = optionsPurchaseQueue;
+        if (_optionsPurchaseQueue != address(0)) {
+            // Allocate a maximum of 50% of options towards the purchase queue
+            uint256 allocatedOptions = optionsMintAmount.div(2);
```

Comment:
Write a setter for what % we allocate the options.  I think 50% is too much and better if we can change it in the future than fix it to 10%, 20%, 50%, etc.. Not sure how popular this will be

Reply:
It only allocates up to 50% (see the `allocateOption()` function

Example 1:
Lets say the vault mints 1000 options. On the purchase queue there are a total of 300 options requested for purchase. Only 300 will get allocated to the purchase queue.

Example 2:
Lets say the vault mints 1000 options. On the purchase queue there are a total of 700 options requested for purchase. Only 500 will get allocated to the purchase queue.